### ISSUE TITLE
Enhancement: [no-unnecessary-type-parameters] Add suggestions for simple cases.

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#9663]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unnecessary-type-parameters/

### Description

Add suggestions with the intended fix.

### Fail

```TypeScript
function toUnknown<T>(x: T): unknown {
    return x;
}

function withExtends<T extends number | string>(x: T): unknown {
    return Math.random();
}
```

### Pass

```TypeScript
// replace unconstrained input generic with unknown
function toUnknown(x: unknown): unknown {
   return x;
}

// replace constrained input generic with its constraint type
function withExtends(x: number | string): unknown {
    return Math.random();
}
```

### Additional Info

There are probably more cases where we could make suggestions, too. These are just the first things that came to mind. I think that adding these suggestions would have high value to help the user understood why the rule is reporting, since this rule is quite subtle

**Comment by JoshuaKGoldberg** [comment]
I think this is a great idea. IMO we don't need to have super high coverage for suggestions at first - just a few common cases could be enough to start.

**Comment by abrahamguo** [comment]
Should this be an autofix rather than a suggestion, since it is guaranteed to be a safe suggestion?

**Comment by kirkwaiblinger** [comment]
*Personally*, my intuition is that it would be better as a suggestion. 
- while the type code changes cannot break runtime behavior, they can and do very easily cause type checking to fail
- with a suggestion we get to supply a message that explains the fix operation
- the fixes are potentially complicated in that they may change types in multiple places in the function. If you have your editor configured to fix on save, this might be pretty disruptive
- speculating, but the fix for a given case may not always be unique? Not sure on this point
